src/engine/objects/resolvers/visitors.rs

```diff
@@ -955,14 +1045,27 @@ pub(super) async fn visit_rel_delete_input<T, RequestCtx>(
 ) -> Result<i32, Error>
 where
     RequestCtx: RequestContext,
-    T: Transaction,
+    T: Transaction<RequestCtx>,
 {
     trace!("visit_rel_delete_input called -- src_query_opt: {:#?}, rel_var: {:#?}, input: {:#?}, info.name: {}",
     src_query_opt, rel_var, input, info.name());
 
     let rel_label = rel_var.src().label()?.to_string() + &rel_var.label().to_title_case() + "Rel";
     let input = if let Some(handlers) = context.event_handlers().before_rel_delete(&rel_label) {
-        handlers.iter().try_fold(input, |v, f| f(v))?
+        for f in handlers.iter() {
+            input = f(
+                input,
+                EventFacade::new(
+                    // TODO: where do we get rel_name ?
+                    CrudOperation::DeleteRel(info.name().to_string(), "?".to_string()),
```

Comment:
The RelQueryVar struct has a function to obtain it, so it'd likely be rel_var.label().

Reply:
Fixed.